File: SDKs/Aspose.Cells-Cloud-SDK-For-Ruby/lib/aspose_cells_cloud/models/protect_sheet_parameter.rb
module AsposeCellsCloud
  # 
  class ProtectSheetParameter < BaseObject
    attr_accessor :protection_type, :password, :allow_edit_area, :allow_deleting_column, :allow_deleting_row, :allow_filtering, :allow_formatting_cell, :allow_formatting_column, :allow_formatting_row, :allow_inserting_column, :allow_inserting_hyperlink, :allow_inserting_row, :allow_selecting_locked_cell, :allow_selecting_unlocked_cell, :allow_sorting, :allow_using_pivot_table
    # attribute mapping from ruby-style variable name to JSON key
    def self.attribute_map
      {
        
        # 
        :'protection_type' => :'ProtectionType',
        
        # 
        :'password' => :'Password',
        
        # 
        :'allow_edit_area' => :'AllowEditArea',
        
        # 
        :'allow_deleting_column' => :'AllowDeletingColumn',
        
        # 
        :'allow_deleting_row' => :'AllowDeletingRow',
        
        # 
        :'allow_filtering' => :'AllowFiltering',
        
        # 
        :'allow_formatting_cell' => :'AllowFormattingCell',
        
        # 
        :'allow_formatting_column' => :'AllowFormattingColumn',
        
        # 
        :'allow_formatting_row' => :'AllowFormattingRow',
        
        # 
        :'allow_inserting_column' => :'AllowInsertingColumn',
        
        # 
        :'allow_inserting_hyperlink' => :'AllowInsertingHyperlink',
        
        # 
        :'allow_inserting_row' => :'AllowInsertingRow',
        
        # 
        :'allow_selecting_locked_cell' => :'AllowSelectingLockedCell',
        
        # 
        :'allow_selecting_unlocked_cell' => :'AllowSelectingUnlockedCell',
        
        # 
        :'allow_sorting' => :'AllowSorting',
        
        # 
        :'allow_using_pivot_table' => :'AllowUsingPivotTable'
        
      }
    end

    # attribute type
    def self.swagger_types
      {
        :'protection_type' => :'String',
        :'password' => :'String',
        :'allow_edit_area' => :'Array<String>',
        :'allow_deleting_column' => :'String',
        :'allow_deleting_row' => :'String',
        :'allow_filtering' => :'String',
        :'allow_formatting_cell' => :'String',
        :'allow_formatting_column' => :'String',
        :'allow_formatting_row' => :'String',
        :'allow_inserting_column' => :'String',
        :'allow_inserting_hyperlink' => :'String',
        :'allow_inserting_row' => :'String',
        :'allow_selecting_locked_cell' => :'String',
        :'allow_selecting_unlocked_cell' => :'String',
        :'allow_sorting' => :'String',
        :'allow_using_pivot_table' => :'String'
        
      }
    end

    def initialize(attributes = {})
      return if !attributes.is_a?(Hash) || attributes.empty?

      # convert string to symbol for hash key
      attributes = attributes.inject({}){|memo,(k,v)| memo[k.to_sym] = v; memo}

      
      if attributes[:'ProtectionType']
        self.protection_type = attributes[:'ProtectionType']
      end
      
      if attributes[:'Password']
        self.password = attributes[:'Password']
      end
      
      if attributes[:'AllowEditArea']
        if (value = attributes[:'AllowEditArea']).is_a?(Array)
          self.allow_edit_area = value
        end
      end
      
      if attributes[:'AllowDeletingColumn']
        self.allow_deleting_column = attributes[:'AllowDeletingColumn']
      end
      
      if attributes[:'AllowDeletingRow']
        self.allow_deleting_row = attributes[:'AllowDeletingRow']
      end
      
      if attributes[:'AllowFiltering']
        self.allow_filtering = attributes[:'AllowFiltering']
      end
      
      if attributes[:'AllowFormattingCell']
        self.allow_formatting_cell = attributes[:'AllowFormattingCell']
      end
      
      if attributes[:'AllowFormattingColumn']
        self.allow_formatting_column = attributes[:'AllowFormattingColumn']
      end
      
      if attributes[:'AllowFormattingRow']
        self.allow_formatting_row = attributes[:'AllowFormattingRow']
      end
      
      if attributes[:'AllowInsertingColumn']
        self.allow_inserting_column = attributes[:'AllowInsertingColumn']
      end
      
      if attributes[:'AllowInsertingHyperlink']
        self.allow_inserting_hyperlink = attributes[:'AllowInsertingHyperlink']
      end
      
      if attributes[:'AllowInsertingRow']
        self.allow_inserting_row = attributes[:'AllowInsertingRow']
      end
      
      if attributes[:'AllowSelectingLockedCell']
        self.allow_selecting_locked_cell = attributes[:'AllowSelectingLockedCell']
      end
      
      if attributes[:'AllowSelectingUnlockedCell']
        self.allow_selecting_unlocked_cell = attributes[:'AllowSelectingUnlockedCell']
      end
      
      if attributes[:'AllowSorting']
        self.allow_sorting = attributes[:'AllowSorting']
      end
      
      if attributes[:'AllowUsingPivotTable']
        self.allow_using_pivot_table = attributes[:'AllowUsingPivotTable']
      end
      
    end

  end
end

File: SDKs/Aspose.Cells-Cloud-SDK-for-.NET/src/Com/Aspose/cells/Model/ProtectSheetParameter.cs
using System;
using System.Text;
using System.Collections;
using System.Collections.Generic;

namespace Com.Aspose.Cells.Model {
  public class ProtectSheetParameter {
    public string ProtectionType { get; set; }

    public string Password { get; set; }

    public List<string> AllowEditArea { get; set; }

    public string AllowDeletingColumn { get; set; }

    public string AllowDeletingRow { get; set; }

    public string AllowFiltering { get; set; }

    public string AllowFormattingCell { get; set; }

    public string AllowFormattingColumn { get; set; }

    public string AllowFormattingRow { get; set; }

    public string AllowInsertingColumn { get; set; }

    public string AllowInsertingHyperlink { get; set; }

    public string AllowInsertingRow { get; set; }

    public string AllowSelectingLockedCell { get; set; }

    public string AllowSelectingUnlockedCell { get; set; }

    public string AllowSorting { get; set; }

    public string AllowUsingPivotTable { get; set; }

    public override string ToString()  {
      var sb = new StringBuilder();
      sb.Append("class ProtectSheetParameter {\n");
      sb.Append("  ProtectionType: ").Append(ProtectionType).Append("\n");
      sb.Append("  Password: ").Append(Password).Append("\n");
      sb.Append("  AllowEditArea: ").Append(AllowEditArea).Append("\n");
      sb.Append("  AllowDeletingColumn: ").Append(AllowDeletingColumn).Append("\n");
      sb.Append("  AllowDeletingRow: ").Append(AllowDeletingRow).Append("\n");
      sb.Append("  AllowFiltering: ").Append(AllowFiltering).Append("\n");
      sb.Append("  AllowFormattingCell: ").Append(AllowFormattingCell).Append("\n");
      sb.Append("  AllowFormattingColumn: ").Append(AllowFormattingColumn).Append("\n");
      sb.Append("  AllowFormattingRow: ").Append(AllowFormattingRow).Append("\n");
      sb.Append("  AllowInsertingColumn: ").Append(AllowInsertingColumn).Append("\n");
      sb.Append("  AllowInsertingHyperlink: ").Append(AllowInsertingHyperlink).Append("\n");
      sb.Append("  AllowInsertingRow: ").Append(AllowInsertingRow).Append("\n");
      sb.Append("  AllowSelectingLockedCell: ").Append(AllowSelectingLockedCell).Append("\n");
      sb.Append("  AllowSelectingUnlockedCell: ").Append(AllowSelectingUnlockedCell).Append("\n");
      sb.Append("  AllowSorting: ").Append(AllowSorting).Append("\n");
      sb.Append("  AllowUsingPivotTable: ").Append(AllowUsingPivotTable).Append("\n");
      sb.Append("}\n");
      return sb.ToString();
    }
  }
  }
